tests/Workflow/MoveJatsAssetsTest.php

```diff
@@ -0,0 +1,333 @@
+        $this->assertEmpty($this->filesystem->listContents());
+    }
+
+    /**
+     * @test
+     */
+    public function it_handles_assets() : void
+    {
+        $mover = new MoveJatsAssets('~.+~', $this->filesystem, $this->client);
+
+        $document = FluentDOM::load(
+            <<<XML
+<item xmlns="http://libero.pub" xml:base="http://www.example.com/">
+    <article xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+        <front>
+            <article-meta>
+                <self-uri xlink:href="/assets/article.pdf"/>
+            </article-meta>
+        </front>
+        <body>
+            <graphic xlink:href="assets/figure1.jpg"/>
+            <sec>
+                <fig>
+                    <graphic xlink:href="assets/figure2.jpg"/>
+                    <media mimetype="video" mime-subtype="avi" xlink:href="assets/figure2">
+                        <caption>
+                            <p>
+                                <fig-group>
+                                    <alternatives>
+                                        <supplementary-material mimetype="foo" mime-subtype="bar" 
+                                            xlink:href="assets/something/figure2.txt"/>
+                                    </alternatives>
+                                </fig-group>
+                            </p>
+                        </caption>
+                    </media>
+                    <graphic xlink:href="/assets/figure2.pdf"/>
+                    <alternatives>
+                        <graphic xlink:href="http://www.example.com/assets/figure2.jpg"/>
+                    </alternatives>
+                </fig>
+            </sec>
+        </body>
+    </article>
+</item>
+XML
+        );
+
+        $marking = new Marking();
+        $task = new PutTask('service', ItemId::fromString('id'), ItemVersionNumber::fromInt(1), $document);
+        $transition = new Transition('transition', 'place1', 'place2');
+
+        $event = new Event($task, $marking, $transition);
+
+        $this->mock->save(
+            new Request('GET', 'http://www.example.com/assets/article.pdf'),
+            new Response(200, ['Content-Type' => 'application/pdf'], 'article pdf')
+        );
+
+        $this->mock->save(
+            new Request('GET', 'http://www.example.com/assets/figure1.jpg'),
+            new Response(200, ['Content-Type' => 'image/jpeg;foo=bar'], 'figure1')
+        );
+
+        $this->mock->save(
+            new Request('GET', 'http://www.example.com/assets/figure2.jpg'),
+            new Response(200, ['Content-Type' => 'image/jpeg;foo=bar'], 'figure2 jpeg')
+        );
+
+        $this->mock->save(
+            new Request('GET', 'http://www.example.com/assets/figure2'),
+            new Response(200, ['Content-Type' => 'video/avi'], 'figure2 avi')
+        );
+
+        $this->mock->save(
+            new Request('GET', 'http://www.example.com/assets/something/figure2.txt'),
```

Comment:
`www.example.com` could be a more specific hostname describing where this come from (likely not a nice URL oriented to the public)